packages/compass-field-store/src/stores/context.ts

```diff
@@ -1,29 +1,104 @@
 import React, { useMemo } from 'react';
 import type { ReactReduxContextValue, TypedUseSelectorHook } from 'react-redux';
-import { createSelectorHook } from 'react-redux';
-import type { FieldsState } from '../modules';
+import { createSelectorHook, createDispatchHook } from 'react-redux';
+import { changeFields, type ConnectionNamespacesState } from '../modules';
 import type { SchemaFieldSubset } from '../modules/fields';
 import { schemaFieldsToAutocompleteItems } from '../modules/fields';
+import { createServiceLocator } from 'hadron-app-registry';
+import type { Schema } from 'mongodb-schema';
+import parseSchema from 'mongodb-schema';
+import {
+  type ConnectionInfoAccess,
+  connectionInfoAccessLocator,
+  useConnectionInfo,
+  useConnectionInfoAccess,
+} from '@mongodb-js/connection-storage/provider';
 
 export const FieldStoreContext = React.createContext<
-  ReactReduxContextValue<FieldsState>
+  ReactReduxContextValue<ConnectionNamespacesState>
 >(
   // @ts-expect-error react-redux types
   null
 );
 
-const useSelector: TypedUseSelectorHook<FieldsState> =
+const useDispatch = createDispatchHook(FieldStoreContext);
+
+const useSelector: TypedUseSelectorHook<ConnectionNamespacesState> =
   createSelectorHook(FieldStoreContext);
 
+export type FieldStoreService = {
+  updateFieldsFromDocuments(
+    ns: string,
+    documents: Record<string, any>[]
+  ): Promise<void>;
+  updateFieldsFromSchema(ns: string, schema: Schema): void;
+};
+
+export function createFieldStoreService(
+  dispatch: ReturnType<typeof useDispatch>,
+  connectionInfoAccess: ConnectionInfoAccess
+): FieldStoreService {
+  return {
+    async updateFieldsFromDocuments(
+      ns: string,
+      documents: Record<string, any>[]
+    ) {
+      try {
+        const { fields } = await parseSchema(documents);
```

Comment:
This probably should be somehow part of the store logic, a separate action, not this service. In the current implementation you basically have two different ways of modifying state of the store, but they are split between two places in the code, this just makes it a bit harder to discover for people who will not be aware of this detail